doc/port_fec_flr/port_fec_flr.md

```diff
@@ -0,0 +1,172 @@
+- SAI_PORT_STAT_IF_IN_FEC_CORRECTABLE_FRAMES, which represents the number of correctable FEC codewords.
+  - return not support if its not working for an interface
+- SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S0, which represents the number of codewords without any errors.
+  - return not support if its not working for an interface
+
+
+### 4.2 Sai Counters Used
+
+The following redis DB entries will be accessed for the FEC FLR calculations
+
+|Redis DB |Table|Entries|New, RW| Format | Description|
+|--------------|-------------|------------------|--------|----------------|----------------|
+|COUNTER_DB |COUNTERS_PORT_NAME_MAP | oid  |R |string |Name to oid mapping |
+|COUNTER_DB |COUNTERS |SAI_PORT_STAT_IF_IN_FEC_NOT_CORRECTABLE_FRAMES |R |number |Total number of uncorrected codewords |
+|COUNTER_DB |COUNTERS |SAI_PORT_STAT_IF_IN_FEC_CORRECTABLE_FRAMES |R |number |Total number of corrected codewords |
+|COUNTER_DB |COUNTERS |SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S0 |R |number |Total number of codewords without any errors |
+|COUNTER_DB |RATES |FEC_FLR |New, RW| floating |calculated FEC FLR |
+|COUNTER_DB |RATES |SAI_PORT_STAT_IF_IN_FEC_NOT_CORRECTABLE_FRAMES_last |NEW, RW |number |Last uncorrectable codewords |
+|COUNTER_DB |RATES |SAI_PORT_STAT_IF_IN_FEC_CORRECTABLE_FRAMES_last |NEW, RW |number |Last correctable codewords |
+|COUNTER_DB |RATES |SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S0_last |NEW, RW |number |Last codewords without any errors |
+
+
+### 4.3 SAI API
+
+No change in the SAI API. No new SAI object accessed.
+
+### 4.4 FEC interleaving
+With FEC interleaving factor (X) incorporated, As per [IEEE 802.3df Logic Ad Hoc](https://www.ieee802.org/3/df/public/adhoc/logic/22_0630/opsasnick_3df_logic_220630a.pdf) FEC FLR is expressed as
+
+For X=1 (no interleaving), FEC_FLR = 1.125 * CER <br>
+For X=2, FEC_FLR = 2.125 * CER <br>
+For X=4, FEC_FLR = 4.125 * CER
+
+By default we consider "no interleaving" and thus FEC_FLR will be computed as "1.125 * CER".
+
+To include the interleaving factor in FEC_FLR computation, a new SAI port attribute will be needed to retrieve the underlying interleaving factor. We could have a CLI configurable to determine whether the interleaving factor should be considered (in the absence of support for this new SAI attribute).
+
+### 4.5 Calculation Formulas
+
+```
+Step 1: calculate CER per PORT_STAT poll interval (default 1s)
+
+    CER = Uncorrectable FEC codewords / (Uncorrectable FEC codewords + Correctable FEC codewords)
+
+    where, Uncorrectable FEC codewords = SAI_PORT_STAT_IF_IN_FEC_NOT_CORRECTABLE_FRAMES - SAI_PORT_STAT_IF_IN_FEC_NOT_CORRECTABLE_FRAMES_last
+	   Correctable FEC codewords = SAI_PORT_STAT_IF_IN_FEC_CORRECTABLE_FRAMES - SAI_PORT_STAT_IF_IN_FEC_CORRECTABLE_FRAMES_last +
+				       SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S0 - SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S0_last
+
+Step 2: calculate FEC FLR using CER and considering interleaving factor (X)
```

Comment:
@vkjammala-arista I am not quite sure if X can vary from platform to platform or this is something FIXED by IEEE based upon the PMD and PCS speed.